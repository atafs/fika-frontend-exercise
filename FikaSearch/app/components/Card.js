import React from 'react';
import {View, StyleSheet, Image} from 'react-native';

import Text from './Text';
import colors from '../config/colors';

function Card({title, subTitle, imageUrl, genreNames}) {
  return (
    <View style={styles.card}>
      <Image
        style={styles.image}
        source={{uri: `https://image.tmdb.org/t/p/w500/${imageUrl}`}}
      />
      <View style={styles.detailsContainer}>
        <Text style={styles.title} numberOfLines={1}>
          {title}
        </Text>
        <Text style={styles.subTitle} numberOfLines={2}>
          {subTitle}
        </Text>
        <Text style={styles.genres} numberOfLines={2}>
          {genreNames}
        </Text>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  card: {
    borderRadius: 15,
    backgroundColor: colors.white,
    marginBottom: 20,
    overflow: 'hidden',
  },
  detailsContainer: {
    padding: 20,
  },
  genres: {
    color: colors.medium,
    fontSize: 14,
  },
  image: {
    width: '100%',
    height: 100,
  },
  subTitle: {
    color: colors.secondary,
    fontWeight: 'bold',
    marginBottom: 7,
  },
  title: {
    marginBottom: 7,
  },
});

export default Card;
