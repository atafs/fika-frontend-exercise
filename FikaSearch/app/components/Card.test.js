import React from 'react';
import renderer from 'react-test-renderer';
import Card from './Card';

describe(Card.name, () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Card />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
