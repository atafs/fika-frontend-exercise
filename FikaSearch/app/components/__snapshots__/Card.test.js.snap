// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderRadius": 15,
      "marginBottom": 20,
      "overflow": "hidden",
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://image.tmdb.org/t/p/w500/undefined",
      }
    }
    style={
      Object {
        "height": 100,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "padding": 20,
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#0c0c0c",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Object {
            "marginBottom": 7,
          },
        ]
      }
    />
    <Text
      numberOfLines={2}
      style={
        Array [
          Object {
            "color": "#0c0c0c",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Object {
            "color": "#4ecdc4",
            "fontWeight": "bold",
            "marginBottom": 7,
          },
        ]
      }
    />
    <Text
      numberOfLines={2}
      style={
        Array [
          Object {
            "color": "#0c0c0c",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Object {
            "color": "#6e6969",
            "fontSize": 14,
          },
        ]
      }
    />
  </View>
</View>
`;
